worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/WorldEditPlugin.java

```diff
@@ -565,14 +567,15 @@ public BukkitPlayer wrapPlayer(Player player) {
         //FAWE start - Use cache over returning a direct BukkitPlayer
         BukkitPlayer wePlayer = getCachedPlayer(player);
         if (wePlayer == null) {
-            synchronized (player) {
-                wePlayer = getCachedPlayer(player);
-                if (wePlayer == null) {
-                    wePlayer = new BukkitPlayer(this, player);
-                    player.setMetadata("WE", new FixedMetadataValue(this, wePlayer));
-                    return wePlayer;
+            Supplier<BukkitPlayer> task = () -> {
+                BukkitPlayer bukkitPlayer = getCachedPlayer(player);
+                if (bukkitPlayer == null) {
+                    bukkitPlayer = new BukkitPlayer(this, player);
+                    player.setMetadata("WE", new FixedMetadataValue(this, bukkitPlayer));
                 }
-            }
+                return bukkitPlayer;
+            };
+            TaskManager.taskManager().sync(task);
```

Comment:
This can cause deadlocks when the server thread is waiting for the current thread. What's the intention behind it?

Reply:
The task manager method here will run the given task immediately if it's on the main thread

Reply:
Yes but it doesn't know if the main thread is blocked by the current thread